### PR TITLE
Mini-PR: Improve Python particle attribute access CI test

### PR DESCRIPTION
**Pull request by roelof-groenewald**
This PR is a follow-up on [this](https://github.com/ECP-WarpX/WarpX/pull/2320#issuecomment-924170859) comment aimed to improve the value of the Python particle attribute access CI test.

### CODE REVIEW

**Comment by roelof-groenewald** [comment]
@ax3l with this test failing in your recent PR it reminded me that I had an outstanding action item to split the test into two separate ones.

File: Regression/WarpX-tests.ini [hunk 1]
```diff
@@ -2606,7 +2606,23 @@ analysisRoutine = Examples/Modules/ParticleBoundaryProcess/analysis_reflection.p
 buildDir = .
 inputFile = Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py
 runtime_params =
-customRunCmd = python PICMI_inputs_2d.py
+customRunCmd = python PICMI_inputs_2d.py 0
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]: Let's add a named parameter here, so it becomes self-describing here. Please see above :)

File: Regression/WarpX-tests.ini [hunk 2]
```diff
@@ -2606,7 +2606,23 @@ analysisRoutine = Examples/Modules/ParticleBoundaryProcess/analysis_reflection.p
...
+numthreads = 1
+compileTest = 0
+doVis = 0
+analysisRoutine = Examples/Tests/ParticleDataPython/analysis.py
+
+[Python_particle_attr_access_unique]
+buildDir = .
+inputFile = Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py
+runtime_params =
+customRunCmd = python PICMI_inputs_2d.py 1
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]: Please see above :)

File: Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py [hunk 3]
```diff
@@ -1,19 +1,9 @@
...
 
-# split processor 0 into separate communicator from others
-comm_world = MPI.COMM_WORLD
-rank = comm_world.Get_rank()
-if rank == 0:
-    color = 0
-else:
-    color = 1
-new_comm = comm_world.Split(color)
+unique = int(sys.argv[1])
```
Comment (ax3l) [inline Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py]: Let's add the Python arg parser here, which will automatically document & validate the mandatory parameter and allows us to also name it on the command line :)

We can also make this a flag that way, e.g., `--unique`.

File: Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py [hunk 4]
```diff
@@ -131,16 +123,14 @@ def add_particles():
 # simulation run
 ##########################
 
-sim.step(max_steps - 1, mpi_comm=new_comm)
```
Comment (ax3l) [inline Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py:134]: Oh, where did the comm go? :)

Do we still cover external communicators in other tests? :)

Comment (roelof-groenewald) [reply Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py:134]: Yeap that is covered in test `Python_pass_mpi_comm`.

File: Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py (hunk 3 above)
Comment (roelof-groenewald) [reply Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py]: Will do!

File: Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
-rank = comm_world.Get_rank()
-if rank == 0:
-    color = 0
-else:
-    color = 1
-new_comm = comm_world.Split(color)
+# Create the parser and add the argument
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '-u', '--unique', type=int, default=0,
```
Comment (ax3l) [inline Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py]:
```suggestion
    '-u', '--unique', action='store_true',
```

File: Regression/WarpX-tests.ini [hunk 6]
```diff
@@ -2606,7 +2606,23 @@ analysisRoutine = Examples/Modules/ParticleBoundaryProcess/analysis_reflection.p
...
+numthreads = 1
+compileTest = 0
+doVis = 0
+analysisRoutine = Examples/Tests/ParticleDataPython/analysis.py
+
+[Python_particle_attr_access_unique]
+buildDir = .
+inputFile = Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py
+runtime_params =
+customRunCmd = python PICMI_inputs_2d.py --unique 1
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]:
```suggestion
customRunCmd = python PICMI_inputs_2d.py --unique
```

File: Regression/WarpX-tests.ini [hunk 7]
```diff
@@ -2606,7 +2606,23 @@ analysisRoutine = Examples/Modules/ParticleBoundaryProcess/analysis_reflection.p
 buildDir = .
 inputFile = Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py
 runtime_params =
-customRunCmd = python PICMI_inputs_2d.py
+customRunCmd = python PICMI_inputs_2d.py --unique 0
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]:
```suggestion
customRunCmd = python PICMI_inputs_2d.py
```

**Review by ax3l — COMMENTED** [review]
Thanks a lot - let's make this a simple boolean flag :)

https://stackoverflow.com/questions/8259001/python-argparse-command-line-flags-without-arguments

**Review by ax3l — APPROVED** [review]
Thank you! :blush:

**Pull request merged**